opentracing-kafka-streams/src/main/java/io/opentracing/contrib/kafka/streams/TracingKafkaClientSupplier.java

```diff
@@ -14,25 +14,41 @@
 package io.opentracing.contrib.kafka.streams;
 
 import io.opentracing.Tracer;
+import io.opentracing.contrib.kafka.ClientSpanNameProvider;
 import io.opentracing.contrib.kafka.TracingKafkaConsumer;
 import io.opentracing.contrib.kafka.TracingKafkaProducer;
 import java.util.Map;
+import java.util.function.BiFunction;
 
 import org.apache.kafka.clients.admin.AdminClient;
 import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
 import org.apache.kafka.clients.producer.KafkaProducer;
 import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
 import org.apache.kafka.common.serialization.ByteArrayDeserializer;
 import org.apache.kafka.common.serialization.ByteArraySerializer;
 import org.apache.kafka.streams.KafkaClientSupplier;
 
 public class TracingKafkaClientSupplier implements KafkaClientSupplier {
 
   private final Tracer tracer;
+  private final BiFunction<String, ConsumerRecord, String> consumerSpanNameProvider;
+  private final BiFunction<String, ProducerRecord, String> producerSpanNameProvider;
 
   public TracingKafkaClientSupplier(Tracer tracer) {
     this.tracer = tracer;
+    this.consumerSpanNameProvider = ClientSpanNameProvider.CONSUMER_OPERATION_NAME;
+    this.producerSpanNameProvider = ClientSpanNameProvider.PRODUCER_OPERATION_NAME;
+  }
+
+  public TracingKafkaClientSupplier(Tracer tracer,
+                                    BiFunction<String, ConsumerRecord, String> consumerSpanNameProvider,
+                                    BiFunction<String, ProducerRecord, String> producerSpanNameProvider) {
+    this.tracer = tracer;
+    this.consumerSpanNameProvider = consumerSpanNameProvider;
+    this.producerSpanNameProvider = producerSpanNameProvider;
```

Comment:
What if these are null? Should we maintain similar behavior to the other constructor? Or let them be null?